### ISSUE TITLE
Gradle sometimes deadlocking after desugaring enabled in Android project

### ISSUE THREAD

**Issue by ilor** [gradle/gradle#17820]
We've had a relatively stable Android Gradle build start randomly getting stuck after we've enabled desugaring in its config. Specifically, this was on Linux with Gradle 6.5.1 and com.android.tools.build:gradle:4.1.3 and the issue stated appearing after adding `coreLibraryDesugaringEnabled true ... coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.5'` to the project. This may be related to #17812 but I'm not sure and I have not checked on master.

Unfortunately I'm not able to attach the problematic project where this reproduces. I'm hoping the attached jstack dump of the stuck daemon [stuck-gradle-daemon-stacks.txt](https://github.com/gradle/gradle/files/6867494/stuck-gradle-daemon-stacks.txt) are actionable. They do mention desugaring and lint, and multiple threads are waiting on ` org.gradle.internal.resources.DefaultResourceLockCoordinationService.withStateLock` which is similar to #17812.

**Comment by h0tk3y** [comment]
Sorry for the late reply.

We believe this issue is no longer relevant. Please file a new issue if necessary.

***

As this issue has no reproducer, we cannot check if it has been fixed, but we believe that the fix for the relevant issue #17812 might have fixed this one as well.
